### PR TITLE
Update recommendations around disabling ThreadAutoFollow and update GA date to Q2

### PR DESCRIPTION
**Pull request by esethna**
1. Updated recommendations around disabling ThreadAutoFollow if the org isn't agile in their hardware.
2. Updated GA target date from Q1 > Q2 (I might have missed some locations)

@cwarnermm please help with text and formatting review, and if you can add any other locations where we may need to make similar changes. Perhaps we should also add similar notes re ThreadAutoFollow to the "Organizing Conversations with CRT" page

### CODE REVIEW

**Comment by esethna** [comment]
Note: This can be merged right away and isn't dependent on any eng work

File: source/install/self-managed-changelog.md
Comment (amyblais) [inline source/install/self-managed-changelog.md:26]:
```suggestion

IMPORTANT: If you upgrade from a release earlier than v6.2, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

File: source/configure/configuration-settings.rst
Comment (amyblais) [inline source/configure/configuration-settings.rst:6000]:
```suggestion
**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable _``ThreadAutoFollow`` <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`_. 
```

File: source/configure/configuration-settings.rst
Comment (amyblais) [inline source/configure/configuration-settings.rst:6051]:
```suggestion
**False**: All backend operations for Collapsed Reply Threads are disabled and server performance will not be impacted by the feature. Collapsed Reply Threads (``CollapsedThreads``) cannot be enabled if ``ThreadAutoFollow`` is disabled.    
```

File: source/configure/configuration-settings.rst
Comment (cwarnermm) [inline source/configure/configuration-settings.rst:6000]:
```suggestion
**Disabled**: Disables Collapsed Reply Threads front-end functionality. To disable all backend operations and thread tracking for Collapsed Reply Threads, you must also disable the `ThreadAutoFollow <https://docs.mattermost.com/configure/configuration-settings.html#automatically-follow-threads>`__ configuration setting.
```

File: source/configure/configuration-settings.rst
Comment (amyblais) [reply source/configure/configuration-settings.rst:6058]: Will do 👍

**Comment by esethna** [comment]
@cwarnermm I think this can be merged after an approval. Thanks all!

**Review by cwarnermm — APPROVED** [review]

**Pull request merged**
